src/lib.rs

```diff
@@ -1055,7 +1055,15 @@ where
 }
 
 impl Default for Docker {
-    /// Returns a Docker instance connected to a unix socket at [DEFAULT_SOCKET_PATH](DEFAULT_SOCKET_PATH)
+    /// Constructs a new Docker instance for a docker host listening at a url specified by an env var `DOCKER_HOST`,
+    /// falling back to unix socket with path [DEFAULT_SOCKET_PATH](DEFAULT_SOCKET_PATH)) if
+    /// `unix-socket` feature is enabled.
+    ///
+    /// WARNING!
```

Comment:
If you want to be more [rustdoc-idiomatic](https://doc.rust-lang.org/book/ch14-02-publishing-to-crates-io.html#commonly-used-sections), then this section could have a header `# Panics`.